statsmodels/discrete/halton_sequence.py

```diff
@@ -0,0 +1,187 @@
+# -*- coding: utf-8 -*-
+"""
+Halton sequence
+
+Original source:
+Quasi Montecarlo Halton sequence generator
+Created on Mon Jun 17 22:12:21 2013
+Author: Sebastien Paris
+Josef Perktold translation from c
+
+http://www.mathworks.com/matlabcentral/fileexchange/17457-quasi-montecarlo-halton-sequence-generator
+http://jpktd.blogspot.ca/2013/06/quasi-random-numbers-with-halton.html
+"""
+
+#void halton(int dim , int nbpts, double *h  , double *p )
+#{
+#
+#	double lognbpts , d , sum;
+#
+#	int i , j , n , t , b;
+#
+#	static int P[11] = {2 ,3 ,5 , 7 , 11 , 13 , 17 , 19 ,  23 , 29 , 31};
+#
+#
+#	lognbpts = log(nbpts + 1);
+#
+#
+#	for(i = 0 ; i < dim ; i++)
+#
+#	{
+#
+#		b      = P[i];
+#
+#		n      = (int) ceil(lognbpts/log(b));
+#
+#
+#		for(t = 0 ; t < n ; t++)
+#
+#		{
+#			p[t] = pow(b , -(t + 1) );
+#		}
+#
+#
+#		for (j = 0 ; j < nbpts ; j++)
+#
+#		{
+#
+#			d        = j + 1;
+#
+#			sum      = fmod(d , b)*p[0];
+#
+#
+#			for (t = 1 ; t < n ; t++)
+#
+#			{
+#
+#				d        = floor(d/b);
+#
+#				sum     += fmod(d , b)*p[t];
+#
+#			}
+#
+#
+#			h[j*dim + i] = sum;
+#
+#		}
+#
+#	}
+#
+#}
+
+from math import log, floor, ceil, fmod
+import numpy as np
+
+def halton(dim, nbpts):
+    """
+    Halton sequence
+
+    Parameters
+    ----------
+    dim : float
+         width of sequence
+    nbpts : float
+        length of sequence
+
+    Returns
+    -------
+    halton : array (nbpts * dim)
+        halton sequence
+
+    Notes
+    ------
+    Quasi Montecarlo generator
+    primes numbers set to:
+    P = [2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31]
+    """
+
+    h = np.empty(nbpts * dim)
+    h.fill(np.nan)
+    p = np.empty(nbpts)
+    p.fill(np.nan)
+
+    # TODO: provided prime number instead of set it static
```

Comment:
There is an interesting thread here about prime number generation: https://stackoverflow.com/questions/2068372/fastest-way-to-list-all-primes-below-n

Reply:
Thanks for the link
I think we can include a (numpy) version
We can also keep the hard coded list if we don't need more than those, i.e. for lower dimensional problems.

Reply:
Otherwise, I thought about something like this:

```python
big_number = 1000
while 'Not enought primes':
    base = primes_from_2_to(big_number)[:dim]
    if len(base) == dim:
        break
    big_number += 1000
```

Reply:
@josef-pkt Actually, I got this working. Here is the link to the gist (MIT licensed of course): https://gist.github.com/tupui/cea0a91cc127ea3890ac0f002f887bae